x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/ParsedInference.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.inference.aggs;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParseException;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.ParsedAggregation;
+import org.elasticsearch.xpack.core.ml.inference.results.FeatureImportance;
+import org.elasticsearch.xpack.core.ml.inference.results.SingleValueInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.TopClassEntry;
+import org.elasticsearch.xpack.core.ml.inference.results.WarningInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.ClassificationConfig;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static org.elasticsearch.xpack.core.ml.inference.results.SingleValueInferenceResults.FEATURE_IMPORTANCE;
+
+
+/**
+ * There isn't enough information in toXContent representation of the
+ * {@link org.elasticsearch.xpack.core.ml.inference.results.InferenceResults}
+ * objects to fully reconstruct them. In particular, depending on which
+ * fields are written (result value, feature importance) it is not possible to
+ * distinguish between a Regression result and a Classification result.
+ *
+ * This class parses the union all possible fields that may be written by
+ * InferenceResults.
+ *
+ * The warning field is mutually exclusive with all the other fields.
+ */
+public class ParsedInference extends ParsedAggregation {
```

Comment:
I do not see where this is used or referenced.

Reply:
These are usually in the HLRC.

Reply:
But in that case it should be in the HLRC itself.

Reply:
This class is used in `InternalInferenceAggregationTests` which extends `InternalAggregationTestCase`. It is the base test that requires an implementation of `ParsedAggregation`. 

I think the reason the class is in the `src` directory rather than `test` is that as Nik said it is used by the HLRC or at least that is the convention I've followed.

Reply:
👍